drivers/SmartThings/bose/src/init.lua

```diff
@@ -198,6 +198,14 @@ local function device_init(driver, device)
   device:set_field("init_started", true)
   device.log.info_with({ hub_logs = true }, "initializing device")
   local serial_number = bose_utils.get_serial_number(device)
+  -- Carry over DTH discovered ip during migration to enable some communication
+  -- in cases where it takes a long time to rediscover the device on the LAN.
+  if not device:get_field("ip") and device.data and device.data.ip then
+    local nu = require "st.net_utils"
+    local ip = nu.convert_ipv4_hex_to_dotted_decimal(device.data.ip)
+    device:set_field("ip", ip, { persist = true })
```

Comment:
So in this case the ip address field is not set but this will catch it and set the ip address for the device?

Reply:
Yeah, when a device is first added to the driver and initialized it will not have an ip address (even the try_create_device from our disco loop will add the device without an ip and the init handler will rediscover the ip). This detects if the device is being added due to a migration (it has `device.data`) in which case we know it has the ip field populated in device data.